bundles/org.openhab.binding.senechome/src/main/java/org/openhab/binding/senechome/internal/SenecHomeHandler.java

```diff
@@ -0,0 +1,244 @@
+import javax.measure.quantity.Power;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.QuantityType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.senechome.internal.json.SenecHomeResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SenecHomeHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Steven Schwarznau - Initial contribution
+ */
+@NonNullByDefault
+public class SenecHomeHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(SenecHomeHandler.class);
+    private final static String VALUE_TYPE_INT = "u3";
+    private final static String VALUE_TYPE_UNSIGNED_INT = "u8";
+    private final static String VALUE_TYPE_FLOAT = "fl";
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+    private @Nullable PowerLimitationStatusDTO limitationStatus = null;
+    private final @Nullable SenecHomeApi senecHomeApi;
+    private SenecHomeConfigurationDTO config = new SenecHomeConfigurationDTO();
+
+    public SenecHomeHandler(Thing thing, SenecHomeApi senecHomeApi) {
+        super(thing);
+        this.senecHomeApi = senecHomeApi;
+    }
+
+    @Override
+    public void handleRemoval() {
+        stopJobIfRunning();
+        super.handleRemoval();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command == REFRESH) {
+            logger.debug("Refreshing {}", channelUID);
+            refresh();
+        } else {
+            logger.warn("The SenecHome-Binding is a read-only binding and can not handle commands");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        stopJobIfRunning();
+        super.dispose();
+    }
+
+    protected void stopJobIfRunning() {
+        final ScheduledFuture<?> refreshJob = this.refreshJob;
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(SenecHomeConfigurationDTO.class);
+        senecHomeApi.setHostname(config.hostname);
+        refreshJob = scheduler.scheduleWithFixedDelay(this::refresh, 0, config.refreshInterval, TimeUnit.SECONDS);
+    }
```

Comment:
Do you need to reset `limitationStatus` to `null` here?

Reply:
I am not sure. In line 63 it is also already nulled. But can't hurt, so will do 👍